pkg/virt-handler/node-labeller/cpu_plugin.go

```diff
@@ -0,0 +1,180 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"strings"
+)
+
+const (
+	usableNo               string = "no"
+	DefaultMinCPU                 = "Penryn"
+	nodeLabellerVolumePath        = "/var/lib/kubevirt-node-labeller/"
+)
+
+var DefaultObsoleteCPUs = []string{"486",
+	"pentium", "pentium2", "pentium3", "pentiumpro",
+	"coreduo", "n270", "core2duo", "Conroe",
+	"athlon", "phenom", "kvm32", "kvm64",
+	"qemu32", "qemu64"}
+
+// getCPUInfo processes all cpu info data and returns
+// slice of usable cpu models and features.
+func (n *NodeLabeller) getCPUInfo() ([]string, cpuFeatures) {
+	minCPU := n.clusterConfig.GetMinCPUModel()
+	if minCPU == "" {
+		minCPU = DefaultMinCPU
+	}
+
+	obsoleteCPUsx86 := n.clusterConfig.GetObsoleteCPUModels()
+	if obsoleteCPUsx86 == nil {
+		obsoleteCPUsx86 = DefaultObsoleteCPUs
+	}
+
+	basicFeaturesMap := n.cpuInfo.models[minCPU]
+
+	cpus := make([]string, 0)
+	features := make(cpuFeatures)
+
+	for _, model := range n.hostCapabilities.items {
+		isObsolete := false
+		for _, obsoleteModel := range obsoleteCPUsx86 {
+			if obsoleteModel == model {
+				isObsolete = true
+			}
+		}
+		if isObsolete {
+			continue
+		}
+
+		newFeatures := subtractMap(n.cpuInfo.models[model], basicFeaturesMap)
+		for feature := range newFeatures {
+			features[feature] = true
+		}
```

Comment:
I'm not sure why do we need this `subtractMap`
perhaps, you can just directly add the missing features.

Instead of `basicFeaturesMap := n.cpuInfo.models[minCPU]` on line 54, just let this be features: `features := n.cpuInfo.models[minCPU]`
then 
```
for feature := range n.cpuInfo.models[model] {
    if _, exist := features[feature]; !exist {
        features[feature] = true
    }
}
```
What do you think?